src/Microsoft.Android.Sdk.Analysis/Analyzers/CustomApplicationAnalyzer.cs

```diff
@@ -0,0 +1,66 @@
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.Android.Sdk.Analysis;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+[DiagnosticAnalyzer (LanguageNames.CSharp)]
+public class CustomApplicationAnalyzer : DiagnosticAnalyzer
+{
+	private const string AndroidApplication = "Android.App.Application";
+	public const string DiagnosticId = "DNAA0001";
+	private static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor (
+		id: DiagnosticId,
+		title: Resources.DNAA0001_Title,
+		messageFormat: Resources.DNAA0001_MessageFormat,
+		category: "Usage",
+		defaultSeverity: DiagnosticSeverity.Warning,
+		isEnabledByDefault: true
+	);
+
+	public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create (Rule);
+
+	public override void Initialize (AnalysisContext context)
+	{
+		context.ConfigureGeneratedCodeAnalysis (GeneratedCodeAnalysisFlags.None);
+		context.EnableConcurrentExecution ();
+
+		// Register a syntax node action to analyze method declarations
+		context.RegisterSyntaxNodeAction (AnalyzeClass, SyntaxKind.ClassDeclaration);
+	}
+
+	private static void AnalyzeClass (SyntaxNodeAnalysisContext context)
+	{
+		var classDeclarationSyntax = context.Node as ClassDeclarationSyntax;
+		if (classDeclarationSyntax == null)
+			return;
+
+		var classSymbol = context.SemanticModel.GetDeclaredSymbol (classDeclarationSyntax) as INamedTypeSymbol;
+		if (classSymbol == null)
+			return;
+
+		if (!Utilities.IsDerivedFrom (classSymbol, AndroidApplication))
+			return;
+
+		var constructors = classDeclarationSyntax.Members
+			.OfType<ConstructorDeclarationSyntax> ();
+
+		bool foundActivationConstructor = false;
+		foreach (var constructor in constructors) {
+			var parameters = constructor.ParameterList.Parameters;
+			if (parameters.Count != 2)
+				continue;
+			if (parameters [0].Type.ToString () != "IntPtr")
+				continue;
+			if (parameters [1].Type.ToString () != "JniHandleOwnership")
+				continue;
```

Comment:
I think these should also check the namespace, I think there is a `ContainingNamespace.Name` property?
* https://github.com/jonathanpeppers/memory-analyzers/blob/2bc83bcb6ab0fd4673a646f1b7bbb2e574cdeac5/MemoryAnalyzers/MemoryAnalyzers/MemoryAnalyzer.cs#L276-L279

Reply:
Not on ParamterSyntax by the looks of things :(

Reply:
Ok, yeah I think you found it, it's on the Symbol:

![image](https://github.com/user-attachments/assets/4fad8803-1140-4ec0-a3fe-13dcc72d44dc)

^^ copilot typed most of that